File: 206 Reverse Linked List/answer_iterative.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* pres = nullptr;
        while (head) {
            ListNode* next = head -> next;
            head -> next = pres;
            pres = head;
            head = next;
        }
        return pres;
    }
};

File: 206 Reverse Linked List/answer_iterative.py
from typing import Optional

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        pres: ListNode = None
        while head:
            next: ListNode = head.next
            head.next = pres
            pres = head
            head = next
        return pres
            
        